src/linux/system.rs

```diff
@@ -855,17 +855,10 @@ fn copy_from_file(entry: &Path) -> Vec<String> {
 
 fn get_all_data_from_file(file: &mut File, size: usize) -> io::Result<String> {
     use std::io::Seek;
-
-    let mut data = Vec::with_capacity(size);
-    unsafe {
-        data.set_len(size);
-    }
-
+    let mut buf = String::with_capacity(size);
```

Comment:
Sorry to come back on this again but: can you make a comparison with https://doc.rust-lang.org/nightly/std/fs/fn.read.html please? I wonder which one between the two is the fastest. I also wonder if it reads all the file like your new implementation.

Reply:
I'll take a look at that and try this as well: https://doc.rust-lang.org/std/fs/fn.read_to_string.html

Reply:
Great idea!

Reply:
In my tests, seems more or less the same but the present solution being the fastest.

`std::fs::read_to_string` :
```
test bench_refresh_disks           ... bench:       9,627 ns/iter (+/- 282)
test bench_refresh_disks_list      ... bench:     146,895 ns/iter (+/- 61,914)
```

`std::fs::read` :
```
test bench_refresh_disks           ... bench:       9,617 ns/iter (+/- 238)
test bench_refresh_disks_list      ... bench:     146,123 ns/iter (+/- 5,694)
```

Present solution:
```
test bench_refresh_disks           ... bench:       9,576 ns/iter (+/- 179)
test bench_refresh_disks_list      ... bench:     139,766 ns/iter (+/- 4,486)
```

However, when compared to reading `/proc/mounts` but returning no disks from `get_all_disks`

```
test bench_refresh_disks           ... bench:           1 ns/iter (+/- 0)
test bench_refresh_disks_list      ... bench:      35,390 ns/iter (+/- 1,211)
```

This seems to show us that the performance results of this function are dependent on the number of file systems present. Though, I suppose the whole function could be optimized out since now it only returns an empty `vec`.

So I ran two more experiments:

Return all disks:
```
test bench_refresh_disks           ... bench:      85,689 ns/iter (+/- 3,345)
test bench_refresh_disks_list      ... bench:     461,211 ns/iter (+/- 71,771)
```

Return just 1 disk:

```
test bench_refresh_disks           ... bench:         647 ns/iter (+/- 4)
test bench_refresh_disks_list      ... bench:      48,183 ns/iter (+/- 15,501)
```

Number of disks seems to be the primary factor here.

Reply:
Then let's keep it this way. Thanks for investigating!